internal/core/operations/switch_active_version/switch_active_version_executor.go

```diff
@@ -0,0 +1,233 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package switch_active_version
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/operations"
+	"github.com/topfreegames/maestro/internal/core/services/interfaces"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type SwitchActiveVersionExecutor struct {
+	roomManager        interfaces.RoomManager
+	schedulerManager   interfaces.SchedulerManager
+	roomsBeingReplaced *sync.Map
+	newCreatedRooms    map[string][]*game_room.GameRoom
+	locker             sync.Mutex
```

Comment:
@caiordjesus why didn't we use sync.Map here?

Reply:
Would be very hard to append to list in key, I would have to load items from list, typecast as the game_room list, append the value and store again. Maybe it wouldn't work since between the time I load from map to the time I store, the key could've changed. I think it's more readable and maintainable this way